📊 urbanization: ghsl data on population and density of capitals

veronikasamborska1994 opened the pull request.

veronikasamborska1994 commented on dag/urbanization.yml:
ah thanks! yes, I need to archive it now. I left it because I was having some discussions with the source regarding the discrepancies between the two versions but I've now resolved these so will archive!

antea04 commented on snapshots/urbanization/2024-12-02/ghsl_urban_centers.xlsx.dvc:
```diff
@@ -0,0 +1,33 @@
+    # Citation
+    producer: European Commission, Joint Research Centre (JRC)
+    citation_full: |-
+      Center For International Earth Science Information Network-CIESIN-Columbia University. 2018. “Gridded Population of the World, Version 4 (GPWv4): Population Count, Revision 11.” Palisades, NY: NASA Socioeconomic Data and Applications Center (SEDAC). https://doi.org/10.7927/H4JW8BX5
```
In the report (https://human-settlement.emergency.copernicus.eu/documents/GHSL_UCDB_R2024.pdf), they use "only":

"European Commission, Joint Research Centre, Melchiorri, M., Mari Rivero, I., Florio, P., Schiavina, M., Krasnodebska, K.,
Politis, P., Uhl, J., Pesaresi, M., Maffenini, L., Sulis, P., Crippa, M., Guizzardi, D., Pisoni, E., Belis, C., Oom, D., Branco, A., Mwaniki, D., Githira, D.,
Kochulem, E., Tommasi, P., Carioli, A., Ehrlich, D., Kemper, T. and Dijkstra, L., Stats in the City the GHSL Urban Centre Database 2025,
Publications Office of the European Union, Luxembourg, 2024, https://data.europa.eu/doi/10.2760/3046391, JRC139768"

But your citation seems more comprehensive, so happy to go with that!

veronikasamborska1994 replied:
that's what the source recommended via email so I'll stick with this one!

antea04 commented on snapshots/urbanization/2024-12-02/ghsl_urban_centers.py:
```diff
@@ -0,0 +1,25 @@
+"""Script to create a snapshot of dataset. This version of the dataset was provided directly by the source via email."""
```
Might be worth to leave the e-mail of the contact here (or in the .dvc), so we can reach out to them directly when updating. But up to you!

veronikasamborska1994 commented on etl/steps/data/garden/urbanization/2024-12-02/ghsl_urban_centers.py:
```diff
@@ -0,0 +1,86 @@
+
+    # Split data into estimates and projections.
+    past_estimates = tb[tb["year"] < START_OF_PROJECTIONS].copy()
+    future_projections = tb[tb["year"] >= START_OF_PROJECTIONS - 5].copy()
```
this is a hack for grapher so that the dashed line starts from 2020 rather than 2025 - otherwise there is a weird break in between...

antea04 replied:
Ah that makes so much sense - thanks for the explanation!

veronikasamborska1994 commented:
ha nicely spotted! thanks a lot @antea04! I think they've put 0 for where they have "missing data" (ie it didn't meet their criteria for an urban center).  I've fixed it now and will put a note somewhere on this but this main issue is now fixed! I'll see what I can do about the city annotation!

antea04 reviewed (approved):
Nice, thanks @veronikasamborska1994! Everything looks good to me - I've approved the charts and happy to have you merge this :)

veronikasamborska1994 commented:
thank you @antea04! :)
